apps/ledger-live-mobile/src/families/tron/Confirmation.tsx

```diff
@@ -0,0 +1,309 @@
+import { Trans, useTranslation } from "react-i18next";
+import { Share, TouchableOpacity } from "react-native";
+import QRCode from "react-native-qrcode-svg";
+import { useSelector } from "react-redux";
+import { useTheme } from "styled-components/native";
+import { track, TrackScreen } from "../../analytics";
+import Alert from "../../components/Alert";
+import CopyLink from "../../components/CopyLink";
+import CurrencyIcon from "../../components/CurrencyIcon";
+import NavigationScrollView from "../../components/NavigationScrollView";
+import PreventNativeBack from "../../components/PreventNativeBack";
+import {
+  BaseComposite,
+  StackNavigatorProps
+} from "../../components/RootNavigator/types/helpers";
+import { ReceiveFundsStackParamList } from "../../components/RootNavigator/types/ReceiveFundsNavigator";
+import { urls } from "../../config/urls";
+import { ScreenName } from "../../const";
+import getWindowDimensions from "../../logic/getWindowDimensions";
+import { accountScreenSelector } from "../../reducers/accounts";
+import AdditionalInfoModal from "../../screens/ReceiveFunds/AdditionalInfoModal";
+import ReceiveSecurityModal from "../../screens/ReceiveFunds/ReceiveSecurityModal";
+
+type ScreenProps = BaseComposite<
+  StackNavigatorProps<
+    ReceiveFundsStackParamList,
+    ScreenName.ReceiveConfirmation | ScreenName.ReceiveVerificationConfirmation
+  >
+>;
+
+type Props = {
+  account?: TokenAccount | Account;
+  parentAccount?: Account;
+  readOnlyModeEnabled?: boolean;
+} & ScreenProps;
+
+export default function ReceiveConfirmation({ navigation }: Props) {
+  const route = useRoute<ScreenProps["route"]>();
+  const { account, parentAccount } = useSelector(accountScreenSelector(route));
+
+  return account ? (
+    <ReceiveConfirmationInner
+      navigation={navigation}
+      route={route}
+      account={account as Account | TokenAccount}
+      parentAccount={parentAccount ?? undefined}
+    />
+  ) : null;
+}
+
+function ReceiveConfirmationInner({
+  navigation,
+  route,
+  account,
+  parentAccount,
+}: Props) {
+  const { colors } = useTheme();
+  const { t } = useTranslation();
+  const verified = route.params?.verified ?? false;
+  const [isModalOpened, setIsModalOpened] = useState(true);
+  const [isToastDisplayed, setIsToastDisplayed] = useState(false);
+  const [isVerifiedToastDisplayed, setIsVerifiedToastDisplayed] =
+    useState(verified);
+  const [isAddionalInfoModalOpen, setIsAddionalInfoModalOpen] = useState(false);
+
+  const hideToast = useCallback(() => {
+    setIsToastDisplayed(false);
+  }, []);
+  const hideVerifiedToast = useCallback(() => {
+    setIsVerifiedToastDisplayed(false);
+  }, []);
+
+  const openAdditionalInfoModal = useCallback(() => {
+    track("notification_clicked", {
+      button: "Imported and created account",
+    });
+    setIsAddionalInfoModalOpen(true);
+    hideToast();
+  }, [setIsAddionalInfoModalOpen, hideToast]);
+
+  const closeAdditionalInfoModal = useCallback(() => {
```

Comment:
We should be more careful with how we call `useCallback`, `useMemo` and such as it's sometimes more costly in performance to call it than not. It's not necessary here

Reply:
It's just a copy paste of an existing file, I duplicate it because we want this error to be exclusive to tron, I have no idea what most of contains of the file is

Reply:
@haammar-ledger Yep this could be discussed outside of here but IMO hooks are tools to optimise (meaning making fast code, faster) and not a tool that improves performance (making slow code, fast).

My arguments against overusing hooks are:

- Most of the time perfomance gains of `useCallback` or `useMemo` are basically unnoticeable but we still do function calls which impact our perf
- More verbose codebase
- This leads to the wrong idea that these hooks solve bigger performance issues, like thinking that as long as we wrap everything with `useCallback`, `useMemo` we don't have to care about things that actually have bigger like useless state change, useless rerenders, component composition, etc
- The [react doc](https://reactjs.org/docs/hooks-reference.html#usememo) states that useMemo might not use the memoized values everytime, it's not a guarantee so we should not rely on them by default

Here are some interesting blog posts:

https://kentcdodds.com/blog/usememo-and-usecallback
https://www.joshwcomeau.com/react/usememo-and-usecallback/
https://www.developerway.com/posts/how-to-use-memo-use-callback
https://maxrozen.com/understanding-when-use-usememo